### PR TITLE
Fix tsindex feb2024

### PR DESCRIPTION
**Pull request by chad-earthscope**
### What does this PR do?

Changses to the obspy.clients.filesystem.tsindex module:

* More fixes to sqlalchemy 2.x warnings-turned-errors, in particular the need to use sa.text() for PRAGMA statements
* Remove leap second file handling, unnecessary with current mseedindex
* Fix tests for mseedindex >= 3.0.4 and remove leap second related tests
* Rebrand IRIS -> EarthScope and fix URLs

### Why was it initiated?  Any relevant Issues?

SQLAlchemy failures during testing and new mseedindex program that handles multiple versions of miniSEED.
Note: the mseedindex package in PyPI is now up to date with the current release: https://pypi.org/project/mseedindex/ and likely how ObsPy users will get this program.

### PR Checklist
- [x] Correct base branch selected?  is now installable via PyPI: `master` for new features, `maintenance_...` for bug fixes
- [ ] This PR is not directly related to an existing issue (which has no PR yet).
- [ ] While the PR is still work-in-progress, the `no_ci` label can be added to skip CI builds
- [ ] If the PR is making changes to documentation, docs pages can be built automatically.
      Just add the `build_docs` tag to this PR.
      Docs will be served at [docs.obspy.org/pr/{branch_name}](https://docs.obspy.org/pr/) (do not use master branch).
      Please post a link to the relevant piece of documentation.
- [ ] If all tests including network modules (e.g. `clients.fdsn`) should be tested for the PR,
      just add the `test_network` tag to this PR.
- [x] All tests still pass.
- [ ] Any new features or fixed regressions are covered via new tests.
- [ ] Any new or changed features are fully documented.
- [x] Significant changes have been added to `CHANGELOG.txt` .
- [ ] First time contributors have added your name to `CONTRIBUTORS.txt` .
- [ ] If the changes affect any plotting functions you have checked that the plots
      from all the CI builds look correct. Add the "upload_plots" tag so that plotting 
      outputs are attached as artifacts. 
- [ ] New modules, add the module to `CODEOWNERS` with your github handle
- [x] Add the yellow `ready for review` label when you are ready for the PR to be reviewed.

### CODE REVIEW

**Comment by megies** [comment]
@chad-earthscope I think this is mixing changes that can go into `maintenance_1.4.x`..
 - rebranding
 - test fixes

.. with a change that should go into `master` instead:
 - functionality change, removing the leap second handling

Can we split this up?

(I changed the merge target branch to see what is actually going on in here)

File: obspy/clients/filesystem/tsindex.py [hunk 1]
```diff
@@ -1550,16 +1357,14 @@ def _fetch_index_rows(self, query_rows=None, bulk_params=None):
...
-                        (sa.literal(f) != '*', sa.literal(f))
-                        ).label("endtime")
+                    sa.case((sa.literal(e) == '*',
+                             sa.literal('0000-00-00T00:00:00')),
+                            else_=sa.literal(e)
+                            ).label("starttime"),
+                    sa.case((sa.literal(f) == '*',
+                             sa.literal('5000-00-00T00:00:00')),
+                            else_=sa.literal(f)
+                            ).label("endtime")
```
Comment (megies) [inline obspy/clients/filesystem/tsindex.py:1367]: This part is different on the `maintenance_1.4.x` branch, so not sure what needs to be done exactly if it should go into 1.4.1

File: obspy/clients/filesystem/tsindex.py [hunk 2]
```diff
@@ -1734,16 +1532,14 @@ def _fetch_summary_rows(self, query_rows):
...
```
Comment (megies) [inline obspy/clients/filesystem/tsindex.py:1542]: Same here..

**Comment by megies** [comment]
Besides these two parts above, the rest could be cherry-picked cleanly on top of a branch off of `maintenance_1.4.x`

**Comment by megies** [comment]
Cherry picked the rebranding for 1.4.1 and this can go in 1.5.0

**Comment by megies** [comment]
Rebased and force-pushed

**Comment by megies** [comment]
All tests pass after the rebase, so I'm assuming we're good to merge this

**Comment by chad-earthscope** [comment]
Thanks  @megies, I owe you a beverage of your choice.

**Pull request merged**